datalab/datalab_session/data_operations/data_operation.py

```diff
@@ -1,21 +1,43 @@
 from abc import ABC, abstractmethod
 import hashlib
 import json
+import os
+import shutil
+import logging
 
 from django.core.cache import cache
-
+from django.conf import settings
 from datalab.datalab_session.tasks import execute_data_operation
 from datalab.datalab_session.utils.format import Format
 
 CACHE_DURATION = 60 * 60 * 24 * 30  # cache for 30 days
 
+log = logging.getLogger()
+log.setLevel(logging.INFO)
 
 class BaseDataOperation(ABC):
 
     def __init__(self, input_data: dict = None):
         """ The data inputs are passed in in the format described from the wizard_description """
         self.input_data = self._normalize_input_data(input_data)
         self.cache_key = self.generate_cache_key()
+        self.temp = settings.TEMP_FITS_DIR # default fallback
+
+        try:
```

Comment:
Don't put this in the __init__ for the data operation. There are several places in the server where we instantiate a data operation just to get its cache_key from an input set to check if its in the cache, where we never intend to do the operation. This could should just wrap the `operate` method to create and destroy the temp dir I think.